### ISSUE TITLE
Add display format to NumericFieldSettings to control the numeric formatting

### ISSUE THREAD

**Issue by MikeAlhayek** [OrchardCMS/OrchardCore#13563]
### Is your feature request related to a problem? Please describe.
Currently there is no way to control the numeric standard numeric-field format like currency, decimal or others. 

### Describe the solution you'd like
I think we can add new value to the NumericFieldSettings that would allow us to store how to display the numeric. In the settings we would store something like `C2`, `N0`, `F4`, etc. In the UI, maybe all we do is show s drop down menu of the format name like 
- General
- Decimal
- Currency
- Exponential
- Fixed-point
- Number
- Percent
- Round-trip
- Hexadecimal

Then show a number field to indicate the scale to display. If not number provided, we'll use the `Scale` we use for the editor.

### Describe alternatives you've considered
Create custom display template which is cumbersome having this part of the options will allow us to change the setting of the default display template

**Comment by ns8482e** [comment]
IMO format should be based on localization Culture of the site

**Comment by MikeAlhayek** [comment]
How would be be able to say display X field using `C2` or `C0`. or `N0` or `F5` using localization?

**Comment by ns8482e** [comment]
https://learn.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.numberformat?view=net-7.0

**Comment by ns8482e** [comment]
And currently it's displaying based on sites current UI culture

**Comment by MikeAlhayek** [comment]
Yes when you do ToString("C0") it's display it based on the site culture with no decimals.

However, there is not way to identify the format per field.

For example, I may have a field called cost and I want to display it using C2. But I have an Msrp field that I want to display using C0. There is no way to specify how to display a field at a field level.

**Comment by sebastienros** [comment]
Isn't it a "Display Setting", the thing that can be added to any field or part to customize how it's rendered.

**Comment by MikeAlhayek** [comment]
Are you suggesting that we add `NumericFieldDisplaySettings` instead of using `NumericFieldSettings`?

**Comment by sebastienros** [comment]
`NumericFieldDisplaySettings`, ok if it's new

**Comment by rwawr** [comment]
Is there any movement planned on this? I just ran into an issue caused by the switch from Newtonsoft to System.Text.Json in 2.x that's somewhat related.

Issue is that Newtonsoft did some [interesting and arguably incorrect things](https://github.com/JamesNK/Newtonsoft.Json/issues/1726) when serializing/deserializing decimals.  In our application, we're importing some data, setting the value of a `NumericField`, then calling the `Apply` method in `ContentExtensions` to set that field on a `ContentPart`. The value of the field is set to 12, but it ends up being written to the database as 12.0 (which seems to happen when calling `JObject.FromObject` on the `ContentElement`). That appears to be fine in 1.8.2, as Newtonsoft also deserializes 12.0 back to 12. But when I update to 2.x, STJ deserializes 12.0 to 12.0, so all of a sudden numeric fields being displayed in the front end of my site have an extra decimal appended to them. This doesn't appear to be a problem with data added in 2.x, as STJ will serialize and deserialize the value as expected. 12 serializes to 12 deserializes to 12, and 12.0 serializes to 12.0 deserializes to 12.0. But data introduced in 1.8.2 then displayed after the update to 2.x is displayed differently than it was before. 

We had been thinking the `Scale` property of `NumericSettings` affected the display, but apparently that's not the case. That said, it would be very nice to have a setting for formatting a `NumericField` so that we could just adjust that setting for the fields we're concerned about rather than overriding the template for the `NumericField` or figuring out some kind of nasty data migration.

**Comment by MikeAlhayek** [comment]
@rwawr I have not had time to work on it yet. If you like, please submit a PR and ping me and I'll review it. 

Another thing you can do temporary in your project in override the https://github.com/OrchardCMS/OrchardCore/blob/c415813912c59f4d7791e631aae2b533ad289100/src/OrchardCore.Modules/OrchardCore.ContentFields/Views/NumericField.cshtml#L4 view and format based on the scaler .

**Comment by infofromca** [comment]
for Currency, https://github.com/OrchardCMS/OrchardCore.Commerce/issues/678
